### PR TITLE
keep-runtime example improvements

### LINKED ISSUE #569: keep-runtime example improvements
**Description**
The existing example in keep-runtime/src/main.rs requires:

1. fixing - the example as is does not work, as it references a non-existent file (`fixtures/return_1.wasm`).  The available file (`fixtures/return_1.wat`) does actually work fine, as does the equivalent .wasm file (constructed using the WABT project's `wat2wasm` utility).  Plan: add the wasm file to the repo (and corresponding files for the other examples).
2. improving - it suggests a `cargo run` as the example.  While an initial `cargo build` is required, using the binary produced thereafter is simpler than the example given (and shouldn't require rust to be installed on the system).  Plan: provide further examples.

### PR DESCRIPTION
**Pull request by MikeCamel**
Added .wasm files in "fixtures" (generated with "wat2wasm")
Added example in src/main.rs to show use of generated binary

Fix #569

### CODE REVIEW

**Comment by ueno** [comment]
@MikeCamel we had the `.wasm` files in the initial version, but replaced them with the `.wat` files during the review, because wasmtime transparently handles both `.wasm` and `.wat` files.

**Comment by MikeCamel** [comment]
> @MikeCamel we had the `.wasm` files in the initial version, but replaced them with the `.wat` files during the review, because wasmtime transparently handles both `.wasm` and `.wat` files. Maybe the example could directly run the `.wat` file?

I thought that might have been the case.  Having the example showing the .wat file is one option.  However, as it's .wasm files that we'll want to be supporting in the long run (it is the standard binary format, after all, and typically much smaller than the .wat format), I think we should definitely show that we support .wasm.

**Comment by MikeCamel** [comment]
> @MikeCamel We definitely want to avoid committing things in git that not human readable. So perhaps a better approach would be to make the runtime accept either `.wat` or `.wasm` files as input. For example, the first four bytes of a `.wasm` contain the magic number: https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md#high-level-structure You could use that to detect a `.wasm` file and fall back to parsing `.wat`.

I think probably the easiest is to change the original example to use just the .wat examples, and remove the .wasm files.

Now that I have Rust compiling to .wasm, and writing to the local filesystem, I can write some simple examples in Rust, and add those to the repo.  I'm going to revert some of the changes in this PR and create a new issue + PR pair for that.

**Comment by ueno** [comment]
> wasmtime transparently handles both .wasm and .wat files

This is no longer true with the latest wasmtime: the method that takes data now assumes that it is in the binary format. A couple of new methods are added to complement it, though they take `AsRef<Path>` and determine the file type based on the file extensions.

As it's unclear whether we will have filesystem abstraction in Keep, it might actually be a better idea to use the binary format extensively (for tests/examples, we can generate the files at build time).

File: keep-runtime/src/main.rs [hunk 1]
```diff
@@ -18,6 +18,19 @@
 //!         ),
 //!     ]
 //! ```
+//! ## Example build and invocation with binary
+//!
+//! ```console
+//! $ cargo build --release
+//! Finished release [optimized] target(s) in 0.06s
+//! $ RUST_LOG=keep_runtime=info RUST_BACKTRACE=1 ./target/x86_64-unknown-linux-musl/release/
+//!    keep-runtime fixtures/wasi_snapshot1.wat
```
Comment (ueno) [inline keep-runtime/src/main.rs:27]: s/wat/wasm/ ? Note that the `.wasm` file is now generated under `OUT_DIR` at build time.

**Comment by mbestavros** [comment]
Completed as part of other work.

**Pull request closed without merging**